analyze-ms-stats/calc-manuscript-stats.py

```diff
@@ -0,0 +1,126 @@
+    references = json.loads(subprocess.getoutput(referencesCommand))
+    num_ref = len(references)
+
+    return([date, clean_date, num_authors, word_count, num_ref])
+
+def main(args):
+    '''Extract statistics from the output branch log'''
+
+    start_time = time.time()
+    print(f"Using {multiprocessing.cpu_count()} CPUs")
+
+    # Read in list of all commits on this branch
+    with open(args.commit_list, "r") as commitFile:
+        commits = [c.strip() for c in commitFile.read().splitlines()]
+
+    # If this analysis has been run before, load in the list of commits analyzed
+    # and only analyze new commits
+    # Assumes no commits will be added retrospectively (to take advantage of linearity)
+    priorData = None
+    if os.path.exists(args.output_table):
+        priorData = pd.read_csv(args.output_table)
+        oldCommits = priorData["commit"].tolist()
+        priorData = priorData.set_index("commit")
+
+        if len(commits) > len(oldCommits):
+            start_old = commits.index(oldCommits[0])
+            commits = commits[:start_old]
+            print(f"{len(commits)} new commits")
+        else:
+            exit("No new commits")
+
+    # Access the variables.json and references.json files associated with each commit and store in dictionary
+    pool = multiprocessing.Pool(processes=multiprocessing.cpu_count())
+    commitData = dict(zip(commits, pool.map(analyze_commit, commits)))
+    pool.close()
+    pool.join()
+
+    # Turn commitData to df, then flip to be in chronological order
+    growthData = pd.DataFrame.from_dict(commitData, orient="index",
+                                        columns=["Date", "Clean_date", "Authors", "Word Count", "References"])
+
+    # Append onto table of previous commit data, if this exists
+    if priorData is not None:
+        print("add to prior data")
+        growthData = growthData.append(priorData)
+
+    # Cache commit data for future updates
+    growthData.to_csv(args.output_table, index_label="commit")
+    print(f'Wrote {args.output_table}')
+
+    # Prepare data to graph
+    graphData = growthData.set_index("Date")
+    graphData = graphData[::-1]
+
+    # Plot the data
+    axes = graphData.plot(kind='line', linewidth=2, subplots=True)
+    for ax in axes:
+        ax.get_yaxis().set_major_formatter(matplotlib.ticker.FuncFormatter(
+            lambda x, p: format(int(x), ',')))
+        ax.set_ylabel('Count')
+        ax.set_ylim(bottom=0)
+        ax.spines['top'].set_visible(False)
+        ax.spines['right'].set_visible(False)
+        ax.minorticks_off()
+        ax.grid(color="lightgray")
+
+    ax.figure.savefig(args.output_figure + '.png', dpi=300, bbox_inches = "tight")
+    ax.figure.savefig(args.output_figure + '.svg', bbox_inches = "tight")
+
+    print(f'Wrote {args.output_figure}.png and {args.output_figure}.svg')
+
+    # Write json output file
+    manuscript_stats = growthData.iloc[0].to_dict()
```

Comment:
When saving to json, let's name these variables to be easier to use as template variables within the text.  I'm using https://github.com/greenelab/covid19-review/blob/external-resources/csse/csse-stats.json as an example.  In this case, what do you think about:
```
  "stats_author_count": "52",
  "stats_clean_date": "September 2, 2021",
  "stats_iso_date": "2021-09-02",
  "stats_references": "1588",
  "stats_word_count": "134821"
```

Or we could use `growth_stats` as the prefix instead of `stats`.

Then we could use those variables instead of the hard-coded values in
> As of April 30, 2021, there were 50 authors, 1,428 references, and 131,949 words in the documents that make up the project.